mmcv/runner/checkpoint.py

```diff
@@ -221,31 +221,120 @@ def _process_mmcls_checkpoint(checkpoint):
+
+class NativeCheckpointLoader(BaseCheckpointLoader):
+
+    @classmethod
+    def load_checkpoint(cls, filename, map_location):
+        if not osp.isfile(filename):
+            raise IOError(f'{filename} is not a checkpoint file')
+        checkpoint = torch.load(filename, map_location=map_location)
+        return checkpoint
+
+
+class CheckpointLoaderClient:
+    """A general checkpoint loader client to load checkpoint.
+
+    Attributes:
+        _loader (dict[loader]): .
+        _native_checkpoint_handler (:obj:`NativeCheckpointLoader`):
+           The backend object.
     """
-    if filename.startswith('modelzoo://'):
-        warnings.warn('The URL scheme of "modelzoo://" is deprecated, please '
-                      'use "torchvision://" instead')
-        model_urls = get_torchvision_models()
-        model_name = filename[11:]
-        checkpoint = load_url_dist(model_urls[model_name])
-    elif filename.startswith('torchvision://'):
-        model_urls = get_torchvision_models()
-        model_name = filename[14:]
-        checkpoint = load_url_dist(model_urls[model_name])
-    elif filename.startswith('open-mmlab://'):
+
+    _loader = {}
+    _native_checkpoint_loader = NativeCheckpointLoader()
+
+    @classmethod
+    def _register_loader(cls, prefixes, loader, force=False):
+        assert isinstance(prefixes, (list, tuple, str))
+        if isinstance(prefixes, str):
+            prefixes = [prefixes]
+        for prefix in prefixes:
+            if prefix not in cls._loader:
+                cls._loader[prefix] = loader
+                continue
+            if force:
+                cls._loader[prefix] = loader
+            else:
+                raise KeyError(
+                    f'{prefix} is already registered as a storage backend, '
+                    'add "force=True" if you want to override it')
+        # sort
+        cls._loader = OrderedDict(
+            sorted(cls._loader.items(), key=lambda t: t[0], reverse=True))
+
+    @classmethod
+    def register_loader(cls, prefixes, loader=None, force=False):
+
+        if loader is not None:
+            cls._register_loader(prefixes, loader, force=force)
+            return
+
+        def _register(loader_cls):
+            cls._register_loader(prefixes, loader_cls, force=force)
+            return loader_cls
+
+        return _register
+
+    @classmethod
+    def _get_checkpoint_loader(cls, path):
+        """Finds a loader that supports the given path. Falls back to the
+        native loader if no other loader is found.
+
+        Args:
+            path (str): checkpoint path
+
+        Returns:
+            loader (Loader)
+        """
+        for p in cls._loader.keys():
+            if path.startswith(p):
+                return cls._loader[p]
+        return cls._native_checkpoint_loader
+
+    @classmethod
+    def load_checkpoint(cls, filepath, map_location=None):
+        return cls._get_checkpoint_loader(filepath).load_checkpoint(
```

Comment:
Shall we indicate which loader is used to load the checkpoint during loading?

Reply:
It May was not good because print information every time you call